src/task1/gui_broken_panel_filter/gui_sorter_working.py

```diff
@@ -0,0 +1,149 @@
+'''
+This is the GUI sorter for further filtering broken solar panels.
+Author: Aman Bhargava (amanb2000)
+
+-------
+Use: python3 gui_sorter_working_prototype.py --from json_name.json --to out_json_name.json
+	- json_name.json is the file that contains the names of the images in the format
+	stated on the trello board
+	- out_json_name.json is the name of the json file to which you would like the
+	program to output the data to.
+
+	Use right arrow key to "swipe right" on truly damaged solar panels
+	Use left arrow key to "swipe left" on not damaged solar panels
+	If you make a mistake, use the up arrow key to go back.
+-------
+Outputs: a JSON file with a list of the right-swiped images
+
+'''
+
+from tkinter import *
+import sys
+import json
+from PIL import ImageTk, Image
+
+'''
+Plan of action:
+	1. (done)Get a bunch of images in this directory
+	2. (done)Make an external json in the format from the trello with the images names
+	3. (done)Make a navigation system for going back/forward through photos (arrow keys)
+	4. (done)Make an internal data storage system (list of names of good images)
+	5. (done)Create a data export system to JSON (in proper format)
+	6. (done)Find a way to make sure all the images are constrianed to fit the window
+	7. (done)Ensure proper IO for this module
+'''
+
+def getScaledDims(xdim, ydim):
+	if (xdim/1280 > ydim/720):
+		return [1280, int(ydim*(1280/xdim))]
+	else:
+		return [int(xdim*(720/ydim)), 720]
+
+# System arguments 1 and 2: The input and output file names
+json_file_name = sys.argv[2]
+json_out_name = sys.argv[4]
+
+# Opening the JSON file and getting its contents, putting the list of images in
+# img_names
+with open(json_file_name) as f:
+	json_in_data = json.load(f)
+
+img_names = json_in_data["positive"]
+
+counter = 0 # Counter for which image we are on
+
+good_images = [] # List to be outputted of the right-swiped images
+
+root = Tk() # Code for making a GUI
+
+# Setting the image for the first time
+img = ImageTk.PhotoImage(Image.open(img_names[counter]))
+scaleDims = getScaledDims(img.width(), img.height())
+img = ImageTk.PhotoImage( (Image.open(img_names[counter]).resize((scaleDims[0], scaleDims[1]))) )
+photo_label = Label(root, image = img)
+photo_label.pack()
+
+# Function called when we get to the end of the list of images to go through.
+def end_program(final_data):
+	with open(json_out_name, 'w') as outfile:
+		outfile.write("{\"Positive\": ")
+		json.dump(final_data, outfile)
```

Comment:
I never like working with hardcoding datastructures like this. You don't have to change it, but if you want to:

```suggestion
		json.dump({"Positive": final_data}, outfile)
```